src/modules/plotter.py

```diff
@@ -0,0 +1,134 @@
+        - A dictionary with metrics in numpy array format.
+        """
+        metrics = {
+            'agent_metrics': {},
+            'cumulative_return': None,
+            'joint_action_frequencies': None
+        }
+
+        # Initialize arrays for storing metrics
+        for key in self.data[self.mode][0].keys():
+            if key not in ['repeat', 'episode', 'actions', 'rewards', 'joint action frequencies', 'metrics']:
+                metrics['agent_metrics'][key] = np.zeros((self.num_repeats, self.num_episodes, self.num_agents))
+
+        metrics['joint_action_frequencies'] = np.zeros((self.num_repeats, self.num_episodes, self.joint_action_size))
+        metrics['cumulative_return'] = np.zeros((self.num_repeats, self.num_episodes)) if self.mode == 'evaluation' else None
+
+        # Fill metrics
+        for entry in self.data[self.mode]:
+            repeat = entry['repeat']
+            episode = entry['episode']
+
+            for key, value in entry.items():
+                if key in metrics['agent_metrics']:
+                    for agent_idx, agent_value in enumerate(value):
+                        metrics['agent_metrics'][key][repeat, episode, agent_idx] = agent_value
+                elif key == 'joint action frequencies':
+                    metrics['joint_action_frequencies'][repeat, episode] = value
+                elif key == 'metrics':
+                    for metric_key, metric_value in value['metrics'].items():
+                        if isinstance(metric_value, list):
+                            if metric_key not in metrics['agent_metrics']:
+                                metrics['agent_metrics'][metric_key] = np.zeros((self.num_repeats, self.num_episodes, self.num_agents))
+                            for agent_idx, agent_value in enumerate(metric_value):
+                                metrics['agent_metrics'][metric_key][repeat, episode, agent_idx] = agent_value
+                        elif metric_key == 'cumulative return':
+                            metrics['cumulative_return'][repeat, episode] = metric_value
+
+        return metrics
+
+    def plot_metric(self, metric_data, metric_name, xlabel='Episodes', ylabel=None):
+        """
+        Plot a specific metric for each agent.
+
+        Parameters:
+        - metric_data (np.array): The metric data to plot.
+        - metric_name (str): The name of the metric to plot.
+        - title (str): The title of the plot.
+        - xlabel (str): The label for the x-axis.
+        - ylabel (str): The label for the y-axis.
+        """
+        if ylabel is None:
+            ylabel = " ".join(metric_name.split('_')).title()
+
+        plt.style.use('ggplot')
+        plt.figure(figsize=(5, 5))
+
+        # Average across repeats
+        avg_metric = np.mean(metric_data, axis=0)
+        std_metric = np.std(metric_data, axis=1)
+        x_values = np.arange(self.num_episodes)
+
+        if metric_name == 'joint_action_frequencies':
```

Comment:
I should consider pruning some episodes to make graphs more striking when it comes to loss function where the main development occurs early in the training.